sdks/python/apache_beam/utils/counters.py

```diff
@@ -170,7 +170,11 @@ class AccumulatorCombineFnCounter(Counter):
   def __init__(self, name, combine_fn):
     assert isinstance(combine_fn, cy_combiners.AccumulatorCombineFn)
     super(AccumulatorCombineFnCounter, self).__init__(name, combine_fn)
-    self._fast_add_input = self.accumulator.add_input
+    # DistributionAccumulator is c-access(cdef) only
+    if isinstance(combine_fn, cy_combiners.DistributionCounterFn):
+      self._fast_add_input = self.accumulator.get_add_input_fn()
```

Comment:
<!--new_thread; commit:e4f1c6aff932ff009d614b61e79a103184529502; resolved:0-->
I'm curious. Why do you need this? Can't you just have `cdef _add_input` and `@property\ cpdef add_input(self): return self._add_input` or something like that?
Maybe I just misunderstood the implementation though.

Reply:
I just commented on this above, but here _fast_add_input is a Python wrapper for the C function (that still has all the Python overhead). cpdef does this for you; just declare add_input as cpdef and then you don't have to change the code here.